src/components/HistoryTable/HistoryTable.tsx

```diff
@@ -1,113 +1,105 @@
-import { useEffect, useRef, ComponentType, useMemo } from "react";
+import { useEffect, useMemo, useRef } from "react";
 import throttle from "lodash.throttle";
-import AutoSizer from "react-virtualized-auto-sizer";
-import {
-  VariableSizeList as List,
-  ListChildComponentProps,
-} from "react-window";
-import InfiniteLoader from "react-window-infinite-loader";
-import { MainlineCommitsForHistoryQuery } from "gql/generated/types";
+import { Virtuoso, VirtuosoHandle } from "react-virtuoso";
+import { useDimensions } from "hooks/useDimensions";
+import { leaveBreadcrumb } from "utils/errorReporting";
+import { types } from ".";
 import { useHistoryTable } from "./HistoryTableContext";
+import LoadingRow from "./HistoryTableRow/LoadingRow";
 
 interface HistoryTableProps {
   loadMoreItems: () => void;
-  recentlyFetchedCommits: MainlineCommitsForHistoryQuery["mainlineCommits"];
-  children: ComponentType<ListChildComponentProps<any>>;
+  children: ({
+    index,
+    data,
+  }: {
+    index: number;
+    data: types.CommitRowType;
+  }) => React.ReactElement;
+  loading: boolean;
 }
 const HistoryTable: React.VFC<HistoryTableProps> = ({
   loadMoreItems,
-  recentlyFetchedCommits,
   children,
+  loading,
 }) => {
   const {
-    commitCount,
     processedCommitCount,
-    selectedCommit,
-    getItemHeight,
-    ingestNewCommits,
-    isItemLoaded,
-    markSelectedRowVisited,
+    processedCommits,
     onChangeTableWidth,
-    toggleRowSizeAtIndex,
+    selectedCommit,
+    visibleColumns,
   } = useHistoryTable();
 
+  const ref = useRef<HTMLDivElement>(null);
+  const listRef = useRef<VirtuosoHandle>(null);
+  const size = useDimensions(ref);
   const throttledOnChangeTableWidth = useMemo(
     () => throttle(onChangeTableWidth, 400),
     [onChangeTableWidth]
   );
 
-  const listRef = useRef<List>(null);
   useEffect(() => {
-    if (recentlyFetchedCommits) {
-      ingestNewCommits(recentlyFetchedCommits);
+    if (size) {
+      throttledOnChangeTableWidth(size.width);
     }
-    // Remove ingestNewCommits from the effect list to avoid infinite loop
-  }, [recentlyFetchedCommits]); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [size, throttledOnChangeTableWidth]);
 
-  // When we fetch new commits we need to tell react-window to re-render the list and update the heights for each of the rows since they will have changed based off of the new commits
   useEffect(() => {
-    if (processedCommitCount > 0) {
-      if (listRef.current) {
-        listRef.current.resetAfterIndex(0);
-      }
+    if (selectedCommit?.loaded && listRef.current) {
+      leaveBreadcrumb(
+        "scrolling to selected commit",
+        {
+          selectedCommit,
+        },
+        "process"
+      );
+      listRef.current.scrollToIndex(selectedCommit.rowIndex);
```

Comment:
I just tried this out. I don't think that would happen the only scenario where I thought this may happen is on the initial render if the selectedCommit is in the first set of downloaded commits. But I just tried it and it consistently executes.